spec/mysql2/client_spec.rb

```diff
@@ -573,7 +586,7 @@ def run_gc
       end
       expect do
         @client.query("SELECT SLEEP(1)")
-      end.to raise_error(Mysql2::Error, /Lost connection to MySQL server/)
+      end.to raise_error(Mysql2::Error, 'MySQL client is not connected')
```

Comment:
Why the different error text?

Reply:
I think it's because we're validating the connection before executing the query (and it's now failing) and it raises with this error text.